bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -346,7 +348,7 @@ public boolean equals(final Object o) {
         if (counter != objectId.counter) {
             return false;
         }
-        if (timestamp != objectId.timestamp) {
+        if (UnsignedInts.compare(timestamp, objectId.timestamp) != 0) {
```

Comment:
If this is the only usage of UnsignedInts, it's not necessary to add that class, as

    timestamp != objectId.timestamp

will suffice for an equality check.

Reply:
I have removed the `UnsignedInts` class and returned the code to its original form.